src/tests/dictionaryConcepts/container/EditConcept.test.jsx

```diff
@@ -61,9 +62,48 @@ describe('Test suite for dictionary concepts components', () => {
     fetchAllConceptSources: jest.fn(),
     allSources: [mockSource],
     addSelectedAnswers: jest.fn(),
+    selectConfirm: jest.fn(),
     selectedAnswers: [],
     createNewAnswerRow: jest.fn(),
   };
+
+  it('should call selectConfirm', () => {
+    const wrapper = mount(<Router>
+      <EditConcept {...props} />
+    </Router>);
+    const instance = wrapper.find('EditConcept').instance();
+    const spy = jest.spyOn(instance, 'selectConfirm');
+    wrapper.find('.btn-danger').at(2).simulate('click');
+    wrapper.find('Button #generalConfirmButton').simulate('click');
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('should show the cancel modal', () => {
+    const wrapper = mount(<Router>
+      <EditConcept {...props} />
+    </Router>);
+    const instance = wrapper.find('EditConcept').instance();
+    instance.showModal();
+    expect(instance.state.show).toEqual(true);
+  });
+
+  it('should redirect on cancel', () => {
+    const wrapper = mount(<Router>
+      <EditConcept {...props} isChange={false} />
+    </Router>);
+    const instance = wrapper.find('EditConcept').instance();
```

Comment:
Where are you asserting for this test?

Reply:
I have adjusted the test.

Reply:
I fixed the issue, I am asserting the value of "isChange" to be false when the "selectConfirm" method is called.

Reply:
Where are you testing for the redirect?

Reply:
I am testing for the redirect at the point where the instance of the "selectConfirm" method is called, given the value of "isChange" is false.

Reply:
How does that prove it is a redirect?

Reply:
I have adjusted the test.